packages/core/test/integration/sequelize/set-session-variables.test.ts

```diff
@@ -0,0 +1,72 @@
+import { expect } from 'chai';
+import { QueryTypes } from '@sequelize/core';
+import {
+  createSequelizeInstance,
+  disableDatabaseResetForSuite,
+  getTestDialect,
+  prepareTransactionTest,
+  sequelize,
+} from '../support';
+
+const dialectName = getTestDialect();
+
+describe('sequelize.setSessionVariables', () => {
+  if (!['mysql', 'mariadb'].includes(dialectName)) {
+    return;
+  }
+
+  disableDatabaseResetForSuite();
+
+  it(`rejects if no connection or transaction is provided`, async () => {
+    await expect(sequelize.setSessionVariables({ foo: 'bar' }))
+      .to.be.rejectedWith(Error, 'specify either options.transaction or options.connection');
+  });
+
+  it('supports CLS transactions', async () => {
+    const clsSequelize = await prepareTransactionTest(createSequelizeInstance({
+      disableClsTransactions: false,
+    }));
+
+    await clsSequelize.transaction(async () => {
+      await clsSequelize.setSessionVariables({ foo: 'bar' });
+      const [data] = await clsSequelize.query<{ foo: string }>('SELECT @foo as `foo`', { type: QueryTypes.SELECT });
+      expect(data).to.be.ok;
+      expect(data.foo).to.be.equal('bar');
+    });
+  });
+
+  it('supports manual transactions', async () => {
+    const transaction = await sequelize.startUnmanagedTransaction();
+
+    try {
```

Comment:
Don't think we need this try-catch here, do we?

Reply:
It should never happen but if this throws before the call to commit, we'll have an uncommitted transaction, and that can have an impact on subsequent tests